PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,235 @@
+import time
+from tkinter import *
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+CELL_SIZE = 40
+CELL_BORDER = 2
+CELL_BORDER_COLOR = 'black'
+STACK_X0 = 300
+STACK_Y0 =  400
+FONT_SIZE = 20
+VALUE_FONT = ('Helvetica', FONT_SIZE)
+VALUE_COLOR = 'black'
+
+class Stack(VisualizationApp):
+    nextColor = 0
+
+    def __init__(self, size=8, title="Stack", **kwargs):
+        super().__init__(title=title, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.buttons = self.makeButtons()
+
+  
+
+    def __str__(self):
+        return str(self.list)
+
+    # STACK FUNCTIONALITY
+
+    def createIndex(self, index, name=None):
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        x0 = STACK_X0 - CELL_SIZE * 4 // 5
+        x1 = STACK_X0 - CELL_SIZE * 3 // 10
+        y0 = y1 = cell_coords[1] + CELL_SIZE // 2
+        return self.drawArrow(
+            x0, y0, x1, y1, color = VARIABLE_COLOR, font= VARIABLE_FONT)
+
+    def drawArrow(         # Create an index arrow to point at an indexed
+            self, x0, y0, x1, y1, color, font,  name=None): # cell with an optional name label
+        arrow = self.canvas.create_line(
+            x0, y0, x1, y1, arrow="last", fill=color)
+        if name:
+            label = self.canvas.create_text(
+                x0 - CELL_SIZE*2 , y0+ CELL_SIZE/5, text=name, anchor=SW,
+                font=font, fill=color)
+        return (arrow, label) if name else (arrow, )
+
+    def push(self, val):
+        self.cleanUp()
+
+        # draw an index pointing to the last cell
+        indexDisplay = self.createIndex(len(self.list))
+        self.cleanup |= set(indexDisplay)
```

Comment:
The set up of the call environment is different now.  Merging `origin/master` into this issue branch will enable you to change these initial lines of the `push()` method to:
```
    callEnviron = self.createCallEnvironment()
    indexDisplay = self.createIndex(len(self.list))
    callEnviron |= set(indexDisplay)
```